### DISCUSSION TITLE
xhttp stream-one/stream-up Nginx

### DISCUSSION THREAD

**Discussion by xqzr** [XTLS/Xray-core#4131] [Show and tell]
```conf
client_header_timeout 5m;
keepalive_timeout 5m;
grpc_read_timeout 315;
grpc_send_timeout 5m;
 location /<path>/ {
 grpc_pass unix:/dev/shm/xrxh.socket;
 grpc_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
 client_max_body_size 0;
 client_body_timeout 5m;
}
```
自用配置，Nginx 没有 error.log 记录
> grpc_read_timeout 315;
> grpc_send_timeout 5m;

它们也可以放 `location` 里

https://t.me/projectXray/4104641?thread=4104641

**Comment by Benjamin1919** [comment]
没看出来这和 stream-up 的配置有什么区别，难道区别就是 path 后面加 "/" ？
我还以为只有 xhttp+reality 才能用 stream-one ……

**Reply by RPRX** [reply]
stream-up 比 stream-one 兼容性好一些，@xqzr 在群里说 CFT 开了流式上行就能用 stream-up，但还得再开个选项才能用 stream-one（可能是 SSE 伪装的锅？），还看到有个 CDN（忘了啥名字）用 stream-one 会被严重限速，但用 stream-up 不会

**Reply by xqzr** [reply]
> @xqzr 在群里说 CFT 开了流式上行就能用 stream-up

嗯？XQ 有说过吗？
XQ 不使用 cloudflront 哦

**Reply by RPRX** [reply]
> > @xqzr 在群里说 CFT 开了流式上行就能用 stream-up
> 
> 嗯？XQ 有说过吗？ XQ 不使用 cloudflront 哦

那记错人了，应该是那位 O_o
